go/teams/chain.go

```diff
@@ -1507,6 +1518,7 @@ func (t *TeamSigChainPlayer) addInnerLink(
 		if prevState.IsImplicit() {
```

Comment:
Maybe mike is right and we should do away with these implicit-team membership checks and have the server deal with it. You'll still check the mapping when you ID the implicit team. This check is more about helping you not hose the team than security.